eZ/Publish/Core/Search/Solr/Content/Gateway/EndpointResolver/NativeEndpointResolver.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * This file is part of the eZ Publish Kernel package
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace eZ\Publish\Core\Search\Solr\Content\Gateway\EndpointResolver;
+
+use eZ\Publish\Core\Search\Solr\Content\Gateway\EndpointResolver;
+use RuntimeException;
+
+/**
+ * NativeEndpointResolver provides Solr endpoints for a Content translations
+ */
+class NativeEndpointResolver implements EndpointResolver
+{
+    /**
+     * Holds an array of Solr entry point names
+     *
+     * @var string[]
+     */
+    protected $entryPoints;
+
+    /**
+     * Holds a map of Endpoint names, with language codes as keys
+     *
+     * <code>
+     *  array(
+     *      "cro-HR" => "endpoint1",
+     *      "eng-GB" => "endpoint2",
+     *  );
+     * </code>
+     *
+     * @var string[]
+     */
+    protected $endpointMap;
```

Comment:
I guess that's a good point... changed in https://github.com/ezsystems/ezpublish-kernel/commit/1dd531c1a46683c1f7ad160aa803b43010317fc7.